soc/nxp/imxrt/Kconfig

```diff
@@ -59,6 +68,21 @@ config FLEXSPI_CONFIG_BLOCK_OFFSET
 	  sequence (optional), etc. The boot ROM expects FlexSPI configuration
 	  parameter to be presented in serial nor flash.
 
+endif # BOOT_FLEXSPI_NOR
+
+if BOOT_XSPI_NOR
+
+config XSPI_CONFIG_BLOCK_OFFSET
+	hex "XSPI config block offset"
+	default 0x0
+	help
+	  XSPI configuration block consists of parameters regarding specific
+	  flash devices including read command sequence, quad mode enablement
+	  sequence (optional), etc. The boot ROM expects XSPI configuration
+	  parameter to be presented in serial nor flash.
```

Comment:
why is this not in dts?

Reply:
Does this make sense for DTS? It is a software setting for the offset of the XSPI configuration block, which will then be used by the bootrom to configure the XSPI to interface with an external flash device and execute from it

Reply:
There is no sense for dts. However, we need to use this kconfig to control config block offset in linker files to ensure the offset of the original block.

Reply:
@nordicjm Devicetree, as per spec, is meant to be consumed by client program. In this case you can think of the zephyr OS as client program. No software in the Zephyr OS will interact with or consume this config block. It is used by the chip boot ROM, and we only need this offset so we can link it to the proper place in the image in order for the chip to boot. I.e., this config block will be consumed even *before* the zephyr reset handler like `z_arm_reset` runs. I don't think it needs to be in DTS.

Reply:
I guess it falls between the cracks. Will ask for @gmarull's opinion